src/controller/java/templates/ClusterInfo-java.zapt

```diff
@@ -149,6 +307,37 @@ public class ClusterInfoMapping {
      {{/chip_client_clusters}}
      return clusterMap;
   }
+
+ public Map<String, ClusterInfo> getReadAttributeMap(Map<String, ClusterInfo> clusterMap) {
+   {{#chip_client_clusters}}
+      Map<String, CommandInfo> read{{asUpperCamelCase name}}CommandInfo = new LinkedHashMap<>();
+     // read attribute
+       {{#chip_server_cluster_attributes}}
+       Map<String, CommandParameterInfo> read{{asUpperCamelCase ../name}}{{asUpperCamelCase name}}CommandParams = new LinkedHashMap<String, CommandParameterInfo>();
+       CommandInfo read{{asUpperCamelCase ../name}}{{asUpperCamelCase name}}AttributeCommandInfo = new CommandInfo(
```

Comment:
Tracking reads using `CommandInfo` makes sense from a code use perspective, but the naming is a bit weird (since reads are different than commands). Let's discuss an alternative offline

Reply:
Name `CommandInfo` into `InteractionInfo`